pyvortex/python/vortex/dataset.py

```diff
@@ -158,7 +158,7 @@ def to_record_batch_reader(
         del memory_pool
```

Comment:
```suggestion
        del memory_pool
        if batch_size is not None:
            raise ValueError("batch_size is not supported")
```
But this should probably be in the same order as the arguments, so before batch_readahead's similar condition.